data_structures/priority_queue.cpp

```diff
@@ -0,0 +1,125 @@
+/**
+ * \file priority_queue.cpp
+ * \brief [Priority Queue Data Structure]
+ *[(Priority Queue)](https://en.wikipedia.org/wiki/Priority_queue)
+ *\details
+ *A priority queue is a type of queue that arranges elements based on their
+ *priority values. Elements with higher priority values are typically retrieved
+ *before elements with lower priority values. In a priority queue, each element
+ *has a priority value associated with it. When you add an element to the queue,
+ *it is inserted in a position based on its priority value.
+ *\author[AbhirajBose](https://github.com/AbhirajBose)
+ *
+ */
+#include <iostream>
+using namespace std;
+/*A structure for Node*/
+struct Node {      // defining node of a LinkedList
+    int data;      // defining data part of a node
+    int priority;  // defining priority part of a node
+    Node* next;    // Node pointer points to next
+};
```

Comment:
Define your structure above from it. 
One more thing whenever we write code, we try that our code must be self explanatory by choosing appropriate variable name and flow of code, which is fine with your code. So, the comments you have added are not necessary here. This is same thing which @MogomotsiFM was highlighting in the suggestions.